.github/workflows/build.yml

```diff
@@ -40,14 +46,63 @@ jobs:
           path: ${{ env.PRE_COMMIT_CACHE_DIR }}
           key: "${{ runner.os }}-pre-commit-py${{ env.PY_VERSION }}-\
             ${{ hashFiles('**/.pre-commit-config.yaml') }}"
+      - name: Cache curl downloads
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.CURL_CACHE_DIR }}
+          key: "${{ runner.os }}-curl-\
+            terraform-${{ env.TERRAFORM_VERSION }}"
+      - name: Lookup go cache directory
+        id: go-cache
+        run: |
+          echo "::set-output name=dir::$(go env GOCACHE)"
+      - name: Cache go
+        uses: actions/cache@v1
+        with:
+          path: ${{ steps.go-cache.outputs.dir }}
+          key: "${{ runner.os }}-go"
+      - name: Cache Terraform initialization
+        uses: actions/cache@v1
+        with:
+          path: ${{ env.TF_CACHE_DIR }}
+          key: "${{ runner.os }}-terraform-initialization"
+      - name: Install Terraform
+        run: |
+          mkdir -p ${{ env.CURL_CACHE_DIR }}
+          TERRAFORM_ZIP="terraform_${TERRAFORM_VERSION}_linux_amd64.zip"
+          curl --output ${{ env.CURL_CACHE_DIR }}/"${TERRAFORM_ZIP}" \
+            --time-cond ${{ env.CURL_CACHE_DIR }}/"${TERRAFORM_ZIP}" \
+            --location \
+            "https://releases.hashicorp.com/terraform/${TERRAFORM_VERSION}/${TERRAFORM_ZIP}"
+          sudo unzip -d /opt/terraform \
+            ${{ env.CURL_CACHE_DIR }}/"${TERRAFORM_ZIP}"
+          sudo ln -s /opt/terraform/terraform /usr/bin/terraform
+      - name: Install Terraform-docs
+        run: GO111MODULE=on go get github.com/segmentio/terraform-docs
+      - name: Find and initialize Terraform directories
+        run: |
+          for path in [[ $(find . -type f -iname "*.tf" -exec dirname "{}" \; \
+            | sort -u) ]]; do \
+            terraform init -input=false -backend=false "$path"; \
+            done
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install --upgrade \
-                      --requirement requirements-test.txt
+          pip install --upgrade --requirement requirements-test.txt
+      - name: Set up pre-commit hook environments
+        run: pre-commit install-hooks
       - name: Run pre-commit on all files
+        # We run pre-commit here with a custom configuration that has
+        # the terraform-validate hook removed.  This is because
+        # terraform validate cannot currently run without accessing
+        # the remote state because of the way the providers are
+        # defined in providers.tf.  This is something that may be
+        # remedied in the future.  For more information, check out
+        # these two GitHub issues:
+        # * https://github.com/hashicorp/terraform/issues/15895
+        # * https://github.com/hashicorp/terraform/issues/15811
         run: pre-commit run --all-files
-
+                            --config=.github-actions-pre-commit-config.yaml
```

Comment:
Let's standardize this syntax the way @jsf9k did it [here](https://github.com/cisagov/cool-assessment-terraform/blob/develop/.github/workflows/build.yml#L104-L106).

Reply:
Ah, I used [this syntax](https://github.com/cisagov/ansible-role-nessus/blob/c42a575a7683b1f036352aae207d8d3983897c64/.github/workflows/build.yml#L104) as a reference (and had to break the line). Resolved in 022fdf0.

Reply:
It wasn't a problem in `ansible-role-nessus` because we have the [line length check disabled](https://github.com/cisagov/ansible-role-nessus/blob/c42a575a7683b1f036352aae207d8d3983897c64/.yamllint#L28) in our ansible repos.  In general, we should use the syntax I mentioned above.